pkg/cluster/delete.go

```diff
@@ -284,8 +284,15 @@ func (m *manager) Delete(ctx context.Context) error {
 		return err
 	}
 
-	m.log.Printf("deleting resources")
-	err = m.deleteResources(ctx)
+	// Retry logic due to race condition between PLS and nic
+	// TODO: Follow up on false future return
+	for retry := 0; retry < 3; retry++ {
+		err = m.deleteResources(ctx)
+		if err == nil {
+			break
+		}
+		m.log.Errorf("Failed to delete %s", err.Error())
+	}
```

Comment:
I wouldn't delete the log entry 
I'm wondering if using `k8s.io/apimachinery/pkg/util/wait` function `PollImmediateUntil` would be valuable.  We used the same logic `pkg/util/cluster/cluster.go` when fixing up network security groups.

Reply:
Talking internal, agree this is good, example cited: https://github.com/Azure/ARO-RP/blob/81229ae1968ebc9f09be4d14428d19d8dde7cce3/pkg/util/cluster/cluster.go#L506-L513

Reply:
Consensus is to go with the temp retry logic. Please see comments in code for more details.